code/modules/mob/living/carbon/status_procs.dm

```diff
@@ -63,6 +63,43 @@
 		clear_fullscreen("high")
 		clear_alert("high")
 
+/mob/living/carbon/adjust_disgust(amount)
+	var/old_disgust = disgust
+	if(amount>0)
+		disgust = min(disgust+amount, DISGUST_LEVEL_MAXEDOUT)
+		switch(disgust)
```

Comment:
It needs to be organ based instead of mob based in our efforts to tie everything to organs.